pkg/cmd/container/cp_linux.go

```diff
@@ -39,17 +40,22 @@ func Cp(ctx context.Context, client *containerd.Client, options types.ContainerC
 				ctx,
 				client,
 				found.Container,
-				options.Container2Host,
-				options.DestPath,
-				options.SrcPath,
-				options.GOptions.Snapshotter,
-				options.FollowSymLink)
+				options)
 		},
 	}
 	count, err := walker.Walk(ctx, options.ContainerReq)
 
-	if count < 1 {
-		err = fmt.Errorf("could not find container: %s, with error: %w", options.ContainerReq, err)
+	if count == -1 {
+		if err == nil {
+			panic("nil error and count == -1 from ContainerWalker.Walk should never happen")
+		}
+		err = errors.Join(errors.New("unable to copy"), err)
+	} else if count == 0 {
+		if err != nil {
+			err = errors.Join(errors.New("unable to retrieve containers with error"), err)
```

Comment:
same behavior you should panic if `err != nil` with `count == 0`

Reply:
Reading into this now. Thanks.

Reply:
I don't think we should panic here.